base/reduce.jl

```diff
@@ -396,7 +396,11 @@ end
 
 
 ## prod
+"""
+    prod(f, iter)
```

Comment:
If I just split off `Callable`, won't the `Any` method automatically just point to the only other 2-arg method? Or does it not work like fallback methods?